sponsorship_compassion/models/contracts.py

```diff
@@ -1350,3 +1352,27 @@ def migrate_gmc_correspondence_commitment(self):
                     self.gmc_correspondent_commitment_id = commitment.get(
                         "CommitmentID"
                     )
+
+    def fix_inconsistent_SWP_contracts(self) -> list:
+        """
+        [T1924] : Finds the contracts with type Write&Pray ("SWP") and fixes the empty
+        contract lines problem which causes the sponsorships not to appear on
+        mycompassion.ch.
+        Returns:
+            list: fixed, previously inconsistent, contracts
+        """
+
+        inconsistent_contracts = self.search([("type", "=", "SWP")])
+        for contract in inconsistent_contracts:
+            if len(contract.contract_line_ids) == 0:
+                # Fix missing contract lines
+                contract._create_empty_lines_for_correspondence()
+                # Update contracts on the partner side
+                contract.correspondent_id._compute_related_contracts()
```

Comment:
HI @ecino ,
I think I finally understood what could be the real cause of the issue after our discussion of yesterday.
Given that the type of the Write&Pray contracts are correctly set, it is still possible that those contracts do not have any contract lines, which causes them to not appear on mycompassion.
What I propose here is to find all those contracts and update them with default lines.
Do you think my reasoning is correct ? For me, it is the most likely explanation, given also the screenshots which were attached to the initial ticket description (showing the empty contract lines).